dlib/cmake_utils/find_blas.cmake

```diff
@@ -66,16 +66,29 @@ if (UNIX OR MINGW)
    find_package(PkgConfig)
    pkg_check_modules(BLAS_REFERENCE cblas)
    pkg_check_modules(LAPACK_REFERENCE lapack)
-   if (BLAS_REFERENCE_FOUND AND LAPACK_REFERENCE_FOUND)
+
+   SET(CMAKE_REQUIRED_LIBRARIES "cblas")
+   INCLUDE(CheckFunctionExists)
+   CHECK_FUNCTION_EXISTS(cblas_ddot HAVE_CBLAS)
+   SET(CMAKE_REQUIRED_LIBRARIES "${BLAS_REFERENCE_LDFLAGS}")
+   if (BLAS_REFERENCE_FOUND AND LAPACK_REFERENCE_FOUND AND HAVE_CBLAS)
       set(blas_libraries "${BLAS_REFERENCE_LDFLAGS}")
       set(lapack_libraries "${LAPACK_REFERENCE_LDFLAGS}")
       set(blas_found 1)
       set(lapack_found 1)
       set(REQUIRES_LIBS "${REQUIRES_LIBS} cblas lapack")
       message(STATUS "Found BLAS and LAPACK via pkg-config")
       return()
+   else()
+      set(blas_libraries cblas blas)
+      set(lapack_libraries "${LAPACK_REFERENCE_LDFLAGS}")
+      set(blas_found 1)
+      set(lapack_found 1)
+      set(REQUIRES_LIBS "${REQUIRES_LIBS} cblas lapack")
+      message(STATUS "Found BLAS and LAPACK via pkg-config")
+      return()
```

Comment:
Delete this block of code.  Think about it.  If `BLAS_REFERENCE_FOUND AND LAPACK_REFERENCE_FOUND AND HAVE_CBLAS` is false it means we **did not find blas or lapack** so things are not good and we can't go saying we found it.

Reply:
I got the point, Thanks @davisking ! :+1: